### PR TITLE
Change max_bytes_for_level_multiplier to double

### PR DESCRIPTION
**Pull request by bgirard**

### CODE REVIEW

File: util/cf_options.cc [hunk 1]
```diff
@@ -75,18 +75,14 @@ ImmutableCFOptions::ImmutableCFOptions(const ImmutableDBOptions& db_options,
       max_subcompactions(db_options.max_subcompactions) {}
 
 // Multiple two operands. If they overflow, return op1.
```
Comment (bgirard) [inline util/cf_options.cc]: I just noticed this comment which doesn't line up with what I assumed the function did on overflow.

File: util/cf_options.cc [hunk 2]
```diff
@@ -75,18 +75,14 @@ ImmutableCFOptions::ImmutableCFOptions(const ImmutableDBOptions& db_options,
...
-uint64_t MultiplyCheckOverflow(uint64_t op1, int op2) {
-  if (op1 == 0) {
+uint64_t MultiplyCheckOverflow(uint64_t op1, double op2) {
+  if (op1 == 0 || op2 <= 0) {
     return 0;
   }
-  if (op2 <= 0) {
+  if (std::numeric_limits<uint64_t>::max() / op1 < op2) {
     return op1;
```
Comment (bgirard) [inline util/cf_options.cc]: And multiplying by zero is not an overflow and I don't see why it made sense to return op1 here. This function is really weird to me, perhaps I'm missing something.

**Comment by yiwu-arbug** [comment]
LGTM. Please mention the change in HISTORY.md.

**Comment by yiwu-arbug** [comment]
Also please fix travis test failures.

File: util/cf_options.cc (hunk 2 above)
Comment (siying) [inline util/cf_options.cc]: By the way, should use port::kMaxUint64.

**Comment by bgirard** [comment]
Do I create a new 4.14.0 for this change in the HISTORY.md?

**Comment by yiwu-arbug** [comment]
@bgirard add "## Unreleased" like this: https://github.com/facebook/rocksdb/blame/d88dff4ef2052cfb031128da9f4dc3be25872d68/HISTORY.md#L2

**Comment by yiwu-arbug** [comment]
Can you also fix AppVeyor build error? Thanks!

**Comment by bgirard** [comment]
Sorry for the spam, I don't have a windows dev machine and msvc is giving me new compile errors each time.

**Comment by yiwu-arbug** [comment]
@bgirard ah, I guess you can look at the error and update PR to see if AppVeyor passes.

**Comment by yiwu-arbug** [comment]
Looking good! One last thing: can you run `make format` and update the PR? Thanks!

**Comment by bgirard** [comment]
Thanks for the quick review!

FYI running 'make format' is broken. I had to read what it was trying to do and do the following:
```
# Location has moved
wget https://raw.githubusercontent.com/leaningtech/duetto-clang/master/tools/clang-format/clang-format-diff.py
# Not sure if I needed clang-format, or it got it from the step below
brew install clang-format
# depot tools is looking for a chromium tree root
touch .gclient
# It was looking for it in a particular location
mkdir -p buildtools/mac
# Happened to have a local checkout of chromium with the script it was looking for
cp ~/browsers/chromium_moved/src/buildtools/mac/clang-format buildtools/mac/clang-format 
```

**Pull request closed without merging**